### PR TITLE
[Merged by Bors] - chore(dependencies): Fix unused dependencies check

### PR DESCRIPTION
**Pull request by morenol**
It failed in last push to master branch:

https://github.com/infinyon/fluvio/actions/runs/2921183973/attempts/1

### CODE REVIEW

File: crates/fluvio-storage/src/segments.rs [hunk 1]
```diff
@@ -178,14 +178,6 @@ impl SegmentList {
         self.segments.len()
     }
 
-    #[instrument(skip(self))]
```
Comment (sehz) [inline crates/fluvio-storage/src/segments.rs:181]: why was this deleted?

Comment (morenol) [reply crates/fluvio-storage/src/segments.rs:181]: I saw that in the output of the failure:
```rust
warning: associated function `get_total_logs_len` is never used
   --> crates/fluvio-storage/src/segments.rs:182:12
    |
182 |     pub fn get_total_logs_len(&self) -> usize {
    |            ^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```

https://github.com/infinyon/fluvio/actions/runs/2921183973/attempts/1

Comment (sehz) [reply crates/fluvio-storage/src/segments.rs:181]: got it.

**Review by sehz — APPROVED** [review]
LGTM
